### PR TITLE
Publish to Windows Package Manager (WinGet)

### PR DESCRIPTION
**Pull request by vedantmgoyal2009**
This action automatically generates manifests for WinGet Community Repository ([microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs)) and submits them.

Before merging this:
1. Please add a GitHub token with `public_repo` scope as a repository secret and rename the secret name in the workflow.
2. Fork [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs) under @telegramdesktop

### CODE REVIEW

**Comment by ilya-fedin** [comment]
> 1. Please add a GitHub token with `public_repo` scope as a repository secret and rename the secret name in the workflow.

Can't you use https://docs.github.com/en/actions/security-guides/automatic-token-authentication?

**Comment by vedantmgoyal2009** [comment]
@ilya-fedin creating a pull request needs an PAT from an account since GitHub Actions cannot do it.

**Comment by ilya-fedin** [comment]
> creating a pull request needs an PAT from an account since GitHub Actions cannot do it.

That seem to be false as https://github.com/desktop-app/action_code_updater works just fine with github,token

**Comment by vedantmgoyal2009** [comment]
It is because that action creates pull request on the same repository where it is being executed. But here, we need to create pull request on another repository.

**Comment by ilya-fedin** [comment]
> But here, we need to create pull request on another repository.

1. On which one?
2. Why can't the action reside in the repository where the PR should be created?

**Comment by vedantmgoyal2009** [comment]
> > But here, we need to create pull request on another repository.
> 
> 1. On which one?

On [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs) repository (it has been clearly mentioned in the pull request body)

> 2. Why can't the action reside in the repository where the PR should be created?

Because whenever a new release of Telegram Desktop, the action is triggered and manifests for the new version of Telegram Desktop are generated and submitted to winget-pkgs repository.

**Comment by ilya-fedin** [comment]
> it has been clearly mentioned in the pull request body

Well, I haven't thought that 'submit' means 'create a PR there'.

> Because whenever a new release of Telegram Desktop, the action is triggered and manifests for the new version of Telegram Desktop are generated and submitted to winget-pkgs repository.

Oh, I wonder whether it's a good practice to spam them with autogenerated PRs (from organization owner account as I understand), but the final decision is for @john-preston anyway.

> Aren't you familiar with [WinGet](https://docs.microsoft.com/en-us/windows/package-manager/winget/)?

No, I'm a Linux user.

File: .github/workflows/winget.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Publish to WinGet
+on:
+  release:
+    types: [released]
```
Comment (vedantmgoyal2009) [inline .github/workflows/winget.yml:4]: @ilya-fedin The action only gets triggered when a draft/pre-release is converted to a release or a release is published. So, it won't spam PRs.

Comment (ilya-fedin) [reply .github/workflows/winget.yml:4]: I mean, it seems they require testing before submitting PRs and this automatic doesn't do this, so the PRs may be of low quality.

Comment (vedantmgoyal2009) [reply .github/workflows/winget.yml:4]: @ilya-fedin can you please ask @john-preston to review this... I think he ignores mentions from other users except you...

**Comment by vedantmgoyal2009** [comment]
In that case, please ask him if you have access to contact him personally 🙏🏻

**Comment by john-preston** [comment]
@vedantmgoyal2009 I'm not against that.

Does this action import some code from your repository which gets access to that PAT and which can be changed by you at any moment later?

**Comment by vedantmgoyal2009** [comment]
No, the action only gets previous version manifests from winget-pkgs repository, updates the version, installer URLs and pushes them to a new branch on the fork and creates a PR using your PAT (the only thing where PAT is needed i.e. to create a pull request).

**Comment by ilya-fedin** [comment]
> No

But the action uses action from your repo of the latest version. It clearly uses your code and you clearly can change it at any time.

**Comment by vedantmgoyal2009** [comment]
If there're any security concerns, I could change the PR to execute a powershell script which would run on this repository and would do the same thing. Please tell me if it can be done.

However, I don't think (and neither can find a way where) someone can use a github action to get someone's PAT and store it. As far as I know, github scans every repository on every push, and if it finds a token compromised, it automatically revokes the token and notifies the owner via email. I was once testing my code and pushed the commit to github without replacing the token to a variable. I immediately got an email that my token was revoked even though it was my repository was private.

**Comment by vedantmgoyal2009** [comment]
@john-preston @ilya-fedin

**Comment by john-preston** [comment]
@vedantmgoyal2009 If you want and if the vedantmgoyal2009/winget-releaser repository contains the required scripts / actions I can clone it to telegramdesktop organization and you can use it from the telegramdesktop organization instead of your own repository.

**Comment by ilya-fedin** [comment]
@john-preston another option is reference the concrete commit you've reviewed instead of "latest"

**Comment by john-preston** [comment]
@vedantmgoyal2009 Ok, I've forked them both to @telegramdesktop.

**Comment by vedantmgoyal2009** [comment]
> * Please add a GitHub token with `public_repo` scope as a repository secret and rename the secret name in the workflow.

Now, just the token has to be added and then the PR can be merged.

**Comment by john-preston** [comment]
@vedantmgoyal2009 I think I've added it. Is there a way to check or just merge it and we'll see?

**Comment by john-preston** [comment]
Thanks!

**Pull request merged**